job_service/adapter/local_storage.py

```diff
@@ -58,3 +77,18 @@ def get_importable_datasets() -> List[ImportableDataset]:
     if ARCHIVE_DIR.exists():
         datasets += get_datasets_in_directory(ARCHIVE_DIR, is_archived=True)
     return datasets
+
+
+def delete_importable_datasets(dataset_name):
+    if not _validate_dataset_name(dataset_name):
```

Comment:
One more thing here: since `get_importable_datasets()` will only return valid dataset names, the validation in `delete_importable_datasets` function is not necessary

Reply:
Linn pointed out the same thing yesterday 👍🏻 but I do think it is important to validate everything server side. Misuse of the delete endpoint could still happen with the validation in the get endpoint present